File: Data Structures & Algorithms/is-anagram/submission-6.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        int[] freq = new int[26];
        
        for (char c: s.toCharArray()) {
            freq[c - 'a'] += 1;
        }
        for (char c: t.toCharArray()) {
            freq[c - 'a'] -=1;
        }
        for (int count: freq) {
            if (count != 0) return false;
        }
        return true;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-6.cs
public class Solution {
    public List<List<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> map = new Dictionary<string, List<string>>();
        foreach(string s in strs) {
            char[] charArray = s.ToCharArray();
            Array.Sort(charArray);
            string key = new string(charArray);
            if (!map.ContainsKey(key)) {
                map[key] = new List<string>();
            }
            map[key].Add(s);
        }
        return map.Values.ToList();
    }
}
